transient-extras.el

```diff
@@ -0,0 +1,115 @@
+  "Format OBJ's value for display and return the result."
+  (let ((argument (oref obj argument)))
+    (if-let ((value (oref obj value)))
+        (propertize
+         (if (listp value)
+             ;; Should be list of files.
+             (mapconcat (lambda (x)
+                          (file-relative-name
+                           (abbreviate-file-name (string-trim x "\"" "\""))))
+                        value " ")
+           ;; Should be a buffer
+           (prin1-to-string value))
+         'face 'transient-value)
+      (propertize argument 'face 'transient-inactive-value))))
+
+(defun transient-extras-read-file (prompt _initial-input _history)
+  "PROMPT for file name.
+
+Returns a list containing the filename. The file must exist."
+  (list (file-local-name (expand-file-name
+                          (read-file-name prompt nil nil t)))))
+
+(transient-define-argument transient-extras-file-list-or-buffer ()
+  :description "Files"
+  :init-value (lambda (obj)
+                (setf (slot-value obj 'value)
+                      (transient-extras--get-default-file-list-or-buffer)))
+  :class 'transient-extras-files-or-buffer)
+
+
+;;; Switches with mutual exclusion
+
+(defclass transient-extras-exclusive-switch (transient-switches) ()
+  "Class used for mutually exclusive command-line switches.
+Similar to `transient-switches' except it allows choices to
+contain different values and labels. In particular, Each element
+in `choices' is a cons of (value . \"label\") and label is used
+for the display.")
+
+(cl-defmethod transient-infix-read ((obj transient-extras-exclusive-switch))
+  "Cycle through the mutually exclusive switches in `choices'."
+  (let* ((choices (mapcar
+                   (apply-partially #'format (oref obj argument-format))
+                   (mapcar
+                    (lambda (x)
+                                        ; Return car of X if it is a cons, otherwise return X.
+                      (if (consp x) (car x) x))
+                    (oref obj choices)))))
+    (if-let ((value (oref obj value)))
+        (cadr (member value choices))
+      (car choices))))
+
+(cl-defmethod transient-format-value ((obj transient-extras-exclusive-switch))
+  "Format OBJ's value for display and return the result."
+  (with-slots (value argument-format choices) obj
+    (format (concat
+             (mapconcat
+              (lambda (choice)
+                (propertize
+                 (if (consp choice) (cdr choice) choice)
+                 'face
+                 (if (equal (format argument-format
+                                    (if (consp choice) (car choice) choice))
+                            value)
+                     'transient-value
+                   'transient-inactive-value)))
+              choices
+              (propertize "|" 'face 'transient-inactive-value))))))
+
+;; TODO: Maybe use `choices' slot to `transient-init-value'
```

Comment:
I think `transient-init-value` works as expected for `transient-extras-exclusive-switch`.
`transient-extras-exclusive-switch` inherits from `transient-switches` (the latter is also an exclusive switch).
The only difference is that `transient-extras-exclusive-switch` allows for a different label to be displayed for options values (for example displaying `draft/normal/best` instead of `3/4/5`).

Reply:
Does it do it without requiring specification of a regular expression?

Reply:
Yes, same as `transient-switches`.

Reply:
Cool!